TerraformPluginDotNet/ResourceProvider/ResourceRegistry.cs

```diff
@@ -5,16 +5,26 @@ namespace TerraformPluginDotNet.ResourceProvider;
 
 class ResourceRegistry
 {
-    public ResourceRegistry(ISchemaBuilder schemaBulder, IEnumerable<ResourceRegistryRegistration> registrations)
+    public ResourceRegistry(
+        ISchemaBuilder schemaBulder,
+        IEnumerable<ResourceRegistryRegistration> resourceRegistrations,
+        IEnumerable<DataSourceRegistryRegistration> dataSourceRegistrations)
     {
-        foreach (var registration in registrations)
+        foreach (var registration in resourceRegistrations)
         {
             Schemas.Add(registration.ResourceName, schemaBulder.BuildSchema(registration.Type));
             Types.Add(registration.ResourceName, registration.Type);
         }
+        foreach (var registration in dataSourceRegistrations)
+        {
+            DataSchemas.Add(registration.ResourceName, schemaBulder.BuildSchema(registration.Type));
+            Types.Add(registration.ResourceName, registration.Type);
+        }
     }
 
     public Dictionary<string, Schema> Schemas { get; } = new Dictionary<string, Schema>();
 
+    public Dictionary<string, Schema> DataSchemas { get; } = new Dictionary<string, Schema>();
```

Comment:
Perhaps the `Types` for data sources should also be separated out, similar to how you have done for schemas? This would mean resources and data sources could have the same names without causing any issues.